Headers/DebugServer2/Utils/Bits.h

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Utils_Bits_h
+#define __DebugServer2_Utils_Bits_h
+
+#include "DebugServer2/Utils/CompilerSupport.h"
+
+namespace ds2 {
+namespace Utils {
+namespace {
+// We apply builtin bit functions on chunks of sizeof(int) * 4 bits.
+const size_t kChunkSize = sizeof(int) * 4;
+}
+
+static inline unsigned int FFS(unsigned int number) {
+#if __has_builtin(__builtin_ffs)
+  return __builtin_ffs(number);
+#elif defined(_MSC_VER)
+  unsigned long index;
+  if (_BitScanForward(&index, number)) {
+    return index + 1;
+  }
+  return 0;
+#else
+  for (int i = 1; number != 0; i++, number >>= 1) {
+    if (number & 1) {
+      return i;
+    }
+  }
+  return 0;
+#endif
+}
+
+template <typename T> static inline unsigned int FFS(T number) {
+  while (number != 0) {
+    int ffs = FFS(number);
+    if (ffs != 0) {
+      return ffs;
+    }
+    number >>= kChunkSize;
+  }
+  return 0;
+}
+
+static inline unsigned int PopCount(unsigned int number) {
+  int count = 0;
+#if __has_builtin(__builtin_popcount)
+  while (number != 0) {
+    count += __builtin_popcount(number);
+    number >>= kChunkSize;
+  }
+#elif defined(_MSC_VER)
+  count = _CountOneBits(number);
+#else
+  while (number != 0) {
+    count++;
+    number = number & (number - 1);
+  }
+#endif
+  return count;
+}
+
+template <typename T> static inline unsigned int PopCount(T number) {
+  int count = 0;
+  while (number != 0) {
+    count += PopCount(number);
+    number >>= kChunkSize;
```

Comment:
@sas no I don't have any benchmarks, but that goes both ways. I tend to prefer to reuse than reimplement, which is why my feedback. If we want performance, I'll mention there is a binary search implementation to do `FFS`.